shell/common/animator.cc

```diff
@@ -264,6 +264,20 @@ void Animator::AwaitVSync() {
   }
 }
 
+void Animator::ForceSyncFrame() {
+  TRACE_EVENT_ASYNC_BEGIN0("flutter", "Forced Frame Request Pending",
+                           frame_request_number_);
+  regenerate_layer_trees_ = true;
+
+  const fml::TimePoint frame_start_time = fml::TimePoint::Now();
+  const fml::TimePoint frame_target_time = frame_start_time;
+
+  std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder =
+      std::make_unique<FrameTimingsRecorder>();
+  frame_timings_recorder->RecordVsync(frame_start_time, frame_target_time);
+  BeginFrame(std::move(frame_timings_recorder));
```

Comment:
`BeginFrame` won't call the begin frame / draw frame callbacks if the frame pipeline is full. I think that's the desired behavior, could you confirm? If yes, we should call this out on the `dart:ui` docs.

Reply:
In the new doc, I've explicitly said that the frame might or might not be rendered depending on situations and platforms.